### ISSUE TITLE
`try?` / `else` not working correctly with `return` statements

### ISSUE THREAD

**Issue by drkameleon** [arturo-lang/arturo#1294]
**Describe the bug**

Probably related to: https://github.com/arturo-lang/arturo/issues/1247 ?

**To Reproduce**

```red
getCorrectVersion: function [versionString][
    try? [
        return to :version versionString
    ]
    else [
        try? [
            return to :version strip replace versionString "v" ""
        ]
        else [
            return null
        ]
    ]
]

getCorrectVersion "v0.9.83" ; returns `null`
```

This returns `null` no matter what.

However, this works:

```red
getCorrectVersion: function [versionString][
    try? [
        to :version versionString
    ]
    else [
        try? [
            to :version strip replace versionString "v" ""
        ]
        else [
            null
        ]
    ]
]

getCorrectVersion "v0.9.83" ; returns `0.9.83` (:version)
```

**Comment by RickBarretto** [comment]
Well, there is no `try?` anymore. 

- Related: #1304

**Comment by RickBarretto** [comment]
- Fixed by #1304
